test/Microsoft.Sbom.Api.Tests/Executors/FileInfoWriterTests.cs

```diff
@@ -0,0 +1,180 @@
+    {
+        // Arrange
+        var sbomConfigs = new[] { sbomConfigMock.Object };
+        var fileInfo = new InternalSbomFileInfo
+        {
+            Path = "test_file.txt",
+            IsOutsideDropPath = false,
+            FileTypes = new[] { FileType.SPDX },
+            Checksum = new[] { new Checksum { Algorithm = AlgorithmName.SHA256, ChecksumValue = "abc123" } }
+        };
+
+        var fileInfoChannel = Channel.CreateUnbounded<InternalSbomFileInfo>();
+        await fileInfoChannel.Writer.WriteAsync(fileInfo);
+        fileInfoChannel.Writer.Complete();
+
+        // Setup expectations
+        sbomPackageDetailsRecorderMock.Setup(m => m.RecordFileId(It.IsAny<string>()));
+
+        // Act
+        var (result, errors) = testSubject.Write(fileInfoChannel.Reader, sbomConfigs);
+
+        // Assert
+        var resultList = new List<JsonDocWithSerializer>();
+        await foreach (var item in result.ReadAllAsync())
+        {
+            resultList.Add(item);
+        }
+
+        var errorList = new List<FileValidationResult>();
+        await foreach (var error in errors.ReadAllAsync())
+        {
+            errorList.Add(error);
+        }
+
+        // Verify file was written to files section
+        Assert.AreEqual(1, resultList.Count);
+        Assert.AreEqual(0, errorList.Count);
+    }
+
+    [TestMethod]
+    public async Task Write_FileOutsideDropPath_DoesNotWriteToFilesSection()
+    {
+        // Arrange
+        var sbomConfigs = new[] { sbomConfigMock.Object };
+        var fileInfo = new InternalSbomFileInfo
+        {
+            Path = "external/package.spdx.json",
+            IsOutsideDropPath = true,
+            FileTypes = new[] { FileType.SPDX },
+            Checksum = new[] { new Checksum { Algorithm = AlgorithmName.SHA256, ChecksumValue = "def456" } }
+        };
+
+        var fileInfoChannel = Channel.CreateUnbounded<InternalSbomFileInfo>();
+        await fileInfoChannel.Writer.WriteAsync(fileInfo);
+        fileInfoChannel.Writer.Complete();
+
+        // Setup expectations - SPDX file ID should still be recorded
+        sbomPackageDetailsRecorderMock.Setup(m => m.RecordSPDXFileId(It.IsAny<string>()));
+
+        // Act
+        var (result, errors) = testSubject.Write(fileInfoChannel.Reader, sbomConfigs);
+
+        // Assert
+        var resultList = new List<JsonDocWithSerializer>();
+        await foreach (var item in result.ReadAllAsync())
+        {
+            resultList.Add(item);
+        }
+
+        var errorList = new List<FileValidationResult>();
+        await foreach (var error in errors.ReadAllAsync())
+        {
+            errorList.Add(error);
+        }
+
+        // Verify file was NOT written to files section
+        Assert.AreEqual(0, resultList.Count);
+        Assert.AreEqual(0, errorList.Count);
+    }
+
+    [TestMethod]
+    public async Task Write_SpdxFileWithinDropPath_RecordsBothFileIdAndSpdxFileId()
```

Comment:
can you combine this test with Write_FileWithinDropPath_WritesToFilesSection? it seems like they are doing very similar work